### PR TITLE
Add MediaType and Route.produces

### PR DESCRIPTION
**Pull request by mofr**
Swagger is in progress, maybe will make a separate PR

### CODE REVIEW

File: winter/http/media_type.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    TEXT_EVENT_STREAM: 'MediaType' = None
+    TEXT_HTML: 'MediaType' = None
+    TEXT_MARKDOWN: 'MediaType' = None
+    TEXT_PLAIN: 'MediaType' = None
+    TEXT_XML: 'MediaType' = None
+
+    def __init__(self, media_type_str: str):
+        self._type_ = '*'
+        self._subtype = '*'
+        self._parameters: Dict[str, str] = {}
```
Comment (berenda-andrey) [inline winter/http/media_type.py]: I think it is better to use tuple of MediaTypeParameter like so
```

@dataclass(frozen=True)
class MediaTypeParameter:
    name: str
    value: str

def __str__(self):
    return f'{self.name}={self.value}'


```

File: winter/http/media_type.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    @property
+    def parameters(self) -> Dict[str, str]:
+        return self._parameters
+
+    @staticmethod
+    def parse(media_type: str) -> Tuple[str, str, Dict[str, str]]:
+        media_type = media_type.strip()
+        if not media_type:
+            raise InvalidMediaTypeException(media_type, 'Media type must not be empty')
+        parameters = {}
```
Comment (berenda-andrey) [inline winter/http/media_type.py]: parameters = []

File: winter/http/media_type.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        parts = media_type.split(';')
+        for param_str in parts[1:]:
+            key_value = param_str.strip().split('=')
+            if len(key_value) != 2:
+                raise InvalidMediaTypeException(media_type, 'Invalid media type parameter list')
+            parameters[key_value[0]] = key_value[1]
```
Comment (berenda-andrey) [inline winter/http/media_type.py]: parameters.append(MediaTypeParameter(*key_value))

File: winter/http/media_type.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+            if len(full_type_parts) > 2:
+                raise InvalidMediaTypeException(media_type, 'Invalid media type format')
+            type_, subtype = full_type_parts
+        if not type_:
+            raise InvalidMediaTypeException(media_type, 'Empty type is specified')
+        if not subtype:
+            raise InvalidMediaTypeException(media_type, 'Empty subtype is specified')
+        if (type_ == '*') != (subtype == '*'):
+            raise InvalidMediaTypeException(media_type, 'Wildcard is allowed only in */* (all media types)')
+        return type_, subtype, parameters
```
Comment (berenda-andrey) [inline winter/http/media_type.py]: type_, subtype, typle(parameters)

File: winter/http/media_type.py (hunk 1 above)
Comment (mofr) [reply winter/http/media_type.py]: @berenda-andrey MediaType has some pre-defined set of possible parameters, like charset. dict will help to lookup for these fields: `MediaType('test/plain; charset=utf-8').parameters.get('charset')`. Could you provide somw benefits of using `List[MediaTypeParameter]`?

Comment (berenda-andrey) [reply winter/http/media_type.py]: It was immutable and hashable by default
I think one media type have not many parameters and searching in typle is also fast

Comment (berenda-andrey) [reply winter/http/media_type.py]: not list, tuple

File: tests/http/test_media_type.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        'boundary': 'secondary',
+    })),
+])
+def test_valid_media_types(media_type, expected_result):
+    result = MediaType.parse(media_type)
+    assert result == expected_result
+
+
+@pytest.mark.parametrize('media_type, expected_message', [
+    ('abc/*', 'Wildcard is allowed only in */* (all media types)'),
```
Comment (mofr) [inline tests/http/test_media_type.py]: It's a valid media type.

File: winter/http/media_type.py (hunk 4 above)
Comment (berenda-andrey) [reply winter/http/media_type.py]:
```
from types import MappingProxyType

return type_, subtype, MappingProxyType(parameters)
```

Noone can change MappingProxyType

**Pull request merged**